tools/test-setup.sh

```diff
@@ -3,5 +3,8 @@ set -euxo pipefail
 # Used by Zuul CI to perform extra bootstrapping
 
 PYTHON=$(command -v python3 python | head -n1)
-
-sudo $PYTHON -m pip install -U tox
+python3 -m pip --version || {
+    curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
+    sudo python3 get-pip.py
```

Comment:
I'd install it under user instead:
```suggestion
    python3 get-pip.py --user
```

Reply:
@webknjaz Sadly, that is not true, please try it yourself, mainly because zuul tox role expect executable tox in user path, and --user does not add it to path. I tried for many months to get some changes to tox role but I lost hope.